pkg/restapi/vc/operation/operations.go

```diff
@@ -1324,6 +1325,117 @@ func (o *Operation) parseAndVerifyVP(vpBytes []byte) error {
 	return nil
 }
 
+func (o *Operation) queryVault(vaultID, vcID string) ([]string, error) {
+	vcIDMAC, err := o.macCrypto.ComputeMAC([]byte(vcID), o.macKeyHandle)
+	if err != nil {
+		return nil, err
+	}
+
+	vcIDIndexValueEncoded := base64.URLEncoding.EncodeToString(vcIDMAC)
+
+	return o.edvClient.QueryVault(vaultID, &models.Query{
+		Name:  o.vcIDIndexNameEncoded,
+		Value: vcIDIndexValueEncoded,
+	})
+}
+
+func (o *Operation) retrieveCredential(rw http.ResponseWriter, profileName string, docIDs []string) {
+	var retrievedVC []byte
+
+	switch len(docIDs) {
+	case 0:
+		o.writeErrorResponse(rw, http.StatusBadRequest,
+			fmt.Sprintf(`no VC under profile "%s" was found with the given id`, profileName))
+	case 1:
+		splitBySlashes := strings.Split(docIDs[0], `/`)
```

Comment:
can you explain logic here

Reply:
Which logic?

Reply:
I see the confusion. The EDV query vault method returns the full URLs of the document (per the spec), and I forgot to update some of the variable names here,

I've updated the PR now to fix the variable names and I also moved the split by slash stuff into a separate function. It should be more clear now.